raiden/transfer/identifiers.py

```diff
@@ -39,3 +32,37 @@ def validate(self) -> None:
 
         if self.channel_identifier < 0 or self.channel_identifier > constants.UINT256_MAX:
             raise ValueError("channel id is invalid")
+
+    def __str__(self) -> str:
+        return (
+            f"{self.chain_identifier}|{to_checksum_address(self.token_network_address)}|"
+            f"{self.channel_identifier}"
+        )
+
+    @staticmethod
+    def from_string(string: str) -> "CanonicalIdentifier":
```

Comment:
Can you just move this to the field? we want to avoid having any serialization / deserialization related methods in the dataclasses.